### PR TITLE
feat: support IVF_HNSW_SQ

### PR DESCRIPTION
**Pull request by BubbleCal**

### CODE REVIEW

**Comment by BubbleCal** [comment]
leave docs and python to future PRs, or it's too large

File: rust/lancedb/src/index/vector.rs
Comment (westonpace) [inline rust/lancedb/src/index/vector.rs:324]: We can add this in a future PR (when we extend this to node and python perhaps) but we will want these parameters to include:

 * What happens if this is set incorrectly?
 * When should the user adjust the default?

File: rust/lancedb/src/index/vector.rs
Comment (westonpace) [inline rust/lancedb/src/index/vector.rs:228]: Can we use `u64` instead of `usize`?  We can convert to `usize` internally.  I'd rather avoid `usize` in our public API since it is slightly more confusing to think about.

**Review by westonpace — APPROVED** [review]
Some minor suggestions.  I think we should decide on `usize` / `u64` before merging (as changing that would be a breaking change) but the other suggestion about the comments we can figure out later.

File: rust/lancedb/src/index/vector.rs
Comment (BubbleCal) [reply rust/lancedb/src/index/vector.rs:228]: sure, let me change it to `u32` cause we don't need such great value of `u64`

**Pull request merged**
